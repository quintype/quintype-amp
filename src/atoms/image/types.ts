import { Common } from "../common-types";

export interface ImageTypes extends Common {
  src: string;
  alt: string;
  srcset?: string;
  attribution?: string;
}
