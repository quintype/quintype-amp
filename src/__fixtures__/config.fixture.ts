import { Config, PublisherConfig, AMPConfig, OptsTypes } from "../types/config";

export const publisherConfig: PublisherConfig = {
  "cdn-name": "https://thumbor-stg.assettype.com/",
  "text-direction": "ltr",
  "publisher-name": "newslaundry"
};

export const ampConfig: AMPConfig = {
  menu: {
    enabled: true,
    items: [
      {
        title: "Home",
        url: "https://barandbench.com"
      },
      {
        title: "News",
        url: "https://www.barandbench.com/news"
      },
      {
        title: "Dealstreet",
        url: "https://www.barandbench.com/dealstreet"
      },
      {
        title: "Conferences & Paper presentations",
        url: "https://www.barandbench.com/apprentice-lawyer/call-for-papers"
      },
      {
        title: "Recruitment Tracker",
        url: "https://www.barandbench.com/apprentice-lawyer/recruitment-tracker"
      },
      {
        title: "Apprentice Lawyer",
        url: "https://www.barandbench.com/apprentice-lawyer"
      },
      {
        title: "Viewpoint",
        url: "https://www.barandbench.com/view-point"
      },
      {
        title: "Legal Jobs",
        url: "https://www.barandbench.com/legal-jobs"
      }
    ]
  },
  fonts: {
    primary: {
      url: "Open+Sans:300,400,600,700",
      family: '"Open Sans", sans-serif'
    },
    secondary: {
      url: "PT+Serif:400,400italic,700,700italic",
      family: "sans-serif"
    }
  },
  colors: {
    primary: "#294f32",
    secondary: "#763194",
    "footer-background": "#FDBD10",
    "footer-text-color": "#FDBD10",
    "header-background": "#dd0b4d",
    "section-text-color": "#f4e842"
  },
  "logo-url": "https://thumbor-stg.assettype.com/ace/2018-02/f0ba1430-8c21-49a3-a5cf-50c88ce9ec74/martian3.jpg",
  doubleclick: {
    "top-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    },
    "body-ad": {
      width: 300,
      height: 250,
      "unit-path": "/60988533/AMP_Body"
    },
    "bottom-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    }
  },
  "google-client-id-api": false,
  "invalid-elements-strategy": "redirect-to-web-version",
  "google-analytics-tracking-id": "UA-ABCDEFG"
};

export const opts: OptsTypes = {
  slots: {
    story: {
      "top-slot": {
        script: `
          <script async custom-element="amp-date-countdown" src="https://cdn.ampproject.org/v0/amp-date-countdown-0.1.js"></script>
          <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
        `,
        ampHtml: `<amp-date-countdown height="100" class="ts_wrapper" timestamp-seconds="2147483648"
        layout="fixed-height">
        <template type="amp-mustache">
          <div>This is the top slot</div>
          <div>
            {{d}} days, {{h}} hours, {{m}} minutes and {{s}} seconds until <a class="ts_a" href="https://en.wikipedia.org/wiki/Year_2038_problem">Y2K38</a>.
          </div>
        </template>
      </amp-date-countdown>`,
        styles: `.ts_wrapper {
          background-image: linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
          filter: brightness(0.9);
          margin: 15px 0;
          line-height: 50px;
          text-align: center;
          display: block;
          color: white;
        }
        .ts_a {
          color: inherit;
        }`
      },
      "bottom-slot": {
        script: `<script async custom-element="amp-dailymotion" src="https://cdn.ampproject.org/v0/amp-dailymotion-0.1.js"></script>`,
        ampHtml: `<amp-dailymotion data-videoid="x3rdtfy"
          layout="responsive"
          data-ui-highlight="FF4081"
          width="480"
          class="storyTopSlot1"
          height="270">
        </amp-dailymotion>`
      }
    }
  }
};

export const config: Config = {
  publisherConfig,
  ampConfig,
  opts
};
