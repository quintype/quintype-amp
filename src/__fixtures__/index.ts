import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
