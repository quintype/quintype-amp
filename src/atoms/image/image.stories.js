import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Image } from "./image";
import { textStory } from "../../__fixtures__/story.fixture";
import { config } from "../../__fixtures__/config.fixture";
import Layout from "../layout";
import _ from "lodash";

const metadata = _.get(textStory, "hero-image-metadata");
const s3key = _.get(textStory, "hero-image-s3-key");
const aspectRatio = [16, 9];

storiesOf("Image", module)
  .addDecorator((story) => <Layout config={config}>{story()}</Layout>)
  .add("layout responsive", () => {
    return <Image metadata={metadata} slug={s3key} aspectRatio={aspectRatio}></Image>;
  })
  .add("layout fixed-height", () => (
    <Image layout={"fixed-height"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio} height={"100"}></Image>
  ))
  .add("layout fixed-height with focus point at 0,0", () => {
    metadata["focus-point"] = [0, 0];
    return (
      <Image layout={"fixed-height"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio} height={"100"}></Image>
    );
  })
  .add("layout fixed-height with focus point at bottom right", () => {
    metadata["focus-point"] = [5472, 3648];
    return (
      <Image layout={"fixed-height"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio} height={"100"}></Image>
    );
  })
  .add("layout fixed", () => {
    return (
      <Image
        layout={"fixed"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        width={"500"}
        height={"500"}></Image>
    );
  })
  .add("layout nodisplay", () => {
    return <Image layout={"nodisplay"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio}></Image>;
  })
  .add("layout fill", () => {
    return (
      <div style={{ width: "250px", height: "250px", position: "relative" }}>
        <Image layout={"fill"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio}></Image>
      </div>
    );
  });
