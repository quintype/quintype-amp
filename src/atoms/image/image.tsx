import React from "react";
import { ImageTypes } from "./types";

export const Image = ({ src, alt, layout = "responsive", ...rest }: ImageTypes) => {
  return <amp-img src={src} alt={alt} layout={layout} {...rest}></amp-img>;
};

export default Image;
