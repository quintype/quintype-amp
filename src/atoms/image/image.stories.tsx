import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Image } from "./image";
import { textStory, config } from "../../__fixtures__";
import { Layout } from "../layout";
import get from "lodash.get";

const metadata = get(textStory, "hero-image-metadata");
const s3key = get(textStory, "hero-image-s3-key");
const caption = get(textStory, "hero-image-caption");
const aspectRatio = [16, 9];

storiesOf("Image", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("layout responsive", () => {
    return <Image metadata={metadata} slug={s3key} alt={caption} />;
  })
  .add("layout fixed-height", () => (
    <Image
      layout={"fixed-height"}
      metadata={metadata}
      slug={s3key}
      aspectRatio={aspectRatio}
      height={"100"}
      alt={caption}
    />
  ))
  .add("layout fixed-height with focus point at 0,0", () => {
    metadata["focus-point"] = [0, 0];
    return (
      <Image
        layout={"fixed-height"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        height={"100"}
        alt={caption}
      />
    );
  })
  .add("layout fixed-height with focus point at bottom right", () => {
    metadata["focus-point"] = [5472, 3648];
    return (
      <Image
        layout={"fixed-height"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        height={"100"}
        alt={caption}
      />
    );
  })
  .add("layout fixed", () => {
    return (
      <Image
        layout={"fixed"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        width={"500"}
        height={"500"}
        alt={caption}
      />
    );
  })
  .add("layout nodisplay", () => {
    return <Image layout={"nodisplay"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio} alt={caption} />;
  })
  .add("layout fill", () => {
    return (
      <div style={{ width: "250px", height: "250px", position: "relative" }}>
        <Image layout={"fill"} metadata={metadata} slug={s3key} aspectRatio={aspectRatio} alt={caption} />
      </div>
    );
  });
