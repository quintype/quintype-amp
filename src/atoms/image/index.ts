import { Image } from "./image";
export default Image;
export { Image };
