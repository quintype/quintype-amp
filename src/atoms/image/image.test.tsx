import React from "react";
import { shallow } from "enzyme";
import { BaseImage } from "./image";
import get from "lodash.get";
import { textStory, config } from "../../__fixtures__";

describe("Image", () => {
  const metadata = get(textStory, "hero-image-metadata");
  const s3key = get(textStory, "hero-image-s3-key");
  const aspectRatio = [16, 9];

  it("should render amp img with src", () => {
    const wrapper = shallow(
      <BaseImage alt="Sample Image" metadata={metadata} slug={s3key} aspectRatio={aspectRatio} config={config} />
    );
    expect(wrapper.find("amp-img").length).toBe(1);
  });
  it("should render fixed-height layout", () => {
    const wrapper = shallow(
      <BaseImage
        alt="Sample Image"
        layout={"fixed-height"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        height={"100"}
        config={config}
      />
    );
    expect(wrapper.find(`amp-img`).prop("height")).toBe("100");
    expect(wrapper.find(`amp-img`).prop("layout")).toBe("fixed-height");
  });
  it("should render fixed layout", () => {
    const wrapper = shallow(
      <BaseImage
        alt="Sample Image"
        config={config}
        layout={"fixed"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
        width={"500"}
        height={"500"}
      />
    );
    expect(wrapper.find(`amp-img`).prop("height")).toBe("500");
    expect(wrapper.find(`amp-img`).prop("width")).toBe("500");
    expect(wrapper.find(`amp-img`).prop("layout")).toBe("fixed");
  });
  it("should render nodisplay layout", () => {
    const wrapper = shallow(
      <BaseImage
        alt="Sample Image"
        config={config}
        layout={"nodisplay"}
        metadata={metadata}
        slug={s3key}
        aspectRatio={aspectRatio}
      />
    );
    expect(wrapper.find(`amp-img`).prop("layout")).toBe("nodisplay");
  });
});
