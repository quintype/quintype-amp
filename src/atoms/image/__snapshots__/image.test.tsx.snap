// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render amp img with src 1`] = `
<div>
  <amp-img
    alt="Amazing image"
    height="500"
    layout="responsive"
    src="/amazingImage.jpg"
    width="400"
  />
</div>
`;
